src/core/mac/mac.cpp

```diff
@@ -832,8 +832,8 @@ otError Mac::PrepareDataRequest(TxFrame &aFrame)
     SuccessOrExit(error = Get<DataPollSender>().GetPollDestinationAddress(dst));
     VerifyOrExit(!dst.IsNone(), error = OT_ERROR_ABORT);
 
-    fcf = Frame::kFcfFrameMacCmd | Frame::kFcfPanidCompression | Frame::kFcfFrameVersion2006 | Frame::kFcfAckRequest |
-          Frame::kFcfSecurityEnabled;
+    fcf = Frame::kFcfFrameMacCmd | Frame::kFcfPanidCompression | Frame::kFcfAckRequest | Frame::kFcfSecurityEnabled;
+    UpdateFrameControlField(false, fcf);
```

Comment:
```suggestion
    UpdateFrameControlField(/* aIsTimeSync */ false, fcf);
```

Reply:
Alternatively, should `aIsTimeSync` be an enum rather than bool? I assume we are preparing for additional header IE types?

Reply:
For now this method is intended to call once to determine the IE present field and frame version when preparing a new frame, so it needs to take all the information about what IEs would be included as input. In predicable future, we would have CSL IE and whether to have it would be inferred from information in mac(so no other input parameters like `aIsTimeSync` is needed). However, if someday we have another new IE and we may probably need to add another input parameter indicating if it would exist. So I think enum may not be used here.